stats_reporter.go

```diff
@@ -69,26 +69,66 @@ type inMemoryCounter struct {
 	Tags  map[string]string
 }
 
-// getKey converts name+tags into a single string of the form
-// "name|tag1=value1|...|tagN=valueN", where tag names are sorted alphabetically.
-func (r *InMemoryStatsCollector) getKey(name string, tags map[string]string) string {
+// MetricDescr describes a metric with tags
+type MetricDescr struct {
+	Name string
+	Tags map[string]string
+}
+
+// NewMetricDescr creates a new metric descriptor from the name and an even number
+// of keyValues strings treates as alternating key, value pairs.
+func NewMetricDescr(name string, keyValues ...string) MetricDescr {
+	m := MetricDescr{Name: name}
+	if len(keyValues)%2 != 0 {
+		return m.WithTag("error", "uneven_keyValues_count")
```

Comment:
this is primarily a testing facility, where fluent syntax is more important. Returning (, error) breaks the flow without much benefits.